torchgeometry/image/color.py

```diff
@@ -1,21 +1,178 @@
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
+    r = image[..., 0, :, :]
+    g = image[..., 1, :, :]
+    b = image[..., 2, :, :]
+
+    maxc = image.max(-3)[0]
+    minc = image.min(-3)[0]
+
+    v = maxc  # brightness
+
+    deltac = maxc - minc
+    s = deltac / v  # saturation
+
+    # avoid division by zero
+    deltac = torch.where(deltac == 0, torch.ones_like(deltac), deltac)
+
+    rc = (maxc - r) / deltac
+    gc = (maxc - g) / deltac
+    bc = (maxc - b) / deltac
+
+    h = 4.0 + gc - rc
+    h[g == maxc] = 2.0 + rc[g == maxc] - bc[g == maxc]
+    h[r == maxc] = bc[r == maxc] - gc[r == maxc]
+    h[minc == maxc] = 0.0
+
+    h = (h / 6.0) % 1.0
+
+    return torch.stack([h, s, v], dim=-3)
+
+
+class RgbToBgr(nn.Module):
+
+    """convert image from RGB to BGR
+    The image data is assumed to be in the range of (0, 1).
+
+    args:
+        image (torch.tensor): RGB image to be converted to BGR
+
+    returns:
+        torch.tensor: BGR version of the image.
+
+    shape:
+        - image: :math:`(*, 3, h, w)`
+        - output: :math:`(*, 3, h, w)`
+
+    """
+
+    def __init__(self) -> None:
+        super(RgbToBgr, self).__init__()
+
+    def forward(self, image: torch.Tensor) -> torch.Tensor:  # type: ignore
+        return rgb_to_bgr(image)
+
+
+def rgb_to_bgr(image: torch.Tensor) -> torch.Tensor:
+    """Convert a RGB image to BGR.
+
+    Args:
+        input (torch.Tensor): RGB Image to be converted to BGR.
+
+    Returns:
+        torch.Tensor: BGR version of the image.
+    """
+
+    return bgr_to_rgb(image)
+
+
+class BgrToRgb(nn.Module):
+
+    """convert image from BGR to RGB.
+    The image data is assumed to be in the range of (0, 1).
+
+    args:
+        image (torch.tensor): BGR image to be converted to RGB.
+
+    returns:
+        torch.tensor: RGB version of the image.
+
+    shape:
+        - image: :math:`(*, 3, h, w)`
+        - output: :math:`(*, 3, h, w)`
+
+    """
+
+    def __init__(self) -> None:
+        super(BgrToRgb, self).__init__()
+
+    def forward(self, image: torch.Tensor) -> torch.Tensor:  # type: ignore
+        return bgr_to_rgb(image)
+
+
+def bgr_to_rgb(image: torch.Tensor) -> torch.Tensor:
+    """Convert a BGR image to RGB.
+
+    Args:
+        input (torch.Tensor): BGR Image to be converted to RGB.
+
+    Returns:
+        torch.Tensor: RGB version of the image.
+    """
+
+    if not torch.is_tensor(image):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W).Got {}"
+                         .format(image.shape))
+
+    out = image.flip(-3)
```

Comment:
could you put a comment here just saying that by doing this we flip the channels ?